huaweicloud/services/acceptance/identitycenter/resource_huaweicloud_identitycenter_account_assignment_test.go

```diff
@@ -0,0 +1,190 @@
+package identitycenter
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getAccountAssignmentResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getAccountAssignment: Query Identity Center account assignment
+	var (
+		getAccountAssignmentHttpUrl = "v1/instances/{instance_id}/account-assignments"
+		getAccountAssignmentProduct = "identitycenter"
+	)
+	getAccountAssignmentClient, err := cfg.NewServiceClient(getAccountAssignmentProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating Identity Center Client: %s", err)
+	}
+
+	getAccountAssignmentBasePath := getAccountAssignmentClient.Endpoint + getAccountAssignmentHttpUrl
+	getAccountAssignmentBasePath = strings.ReplaceAll(getAccountAssignmentBasePath, "{instance_id}",
+		state.Primary.Attributes["instance_id"])
+
+	getAccountAssignmentPath := getAccountAssignmentBasePath + buildGetAccountAssignmentQueryParams(state, "")
+
+	getAccountAssignmentOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	var accountPermission interface{}
+	principalId := state.Primary.Attributes["principal_id"]
+getAccountPermissionsLoop:
+	for {
+		getAccountAssignmentResp, err := getAccountAssignmentClient.Request("GET", getAccountAssignmentPath,
+			&getAccountAssignmentOpt)
+
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving Identity Center account assignment: %s", err)
+		}
+
+		getAccountAssignmentRespBody, err := utils.FlattenResponse(getAccountAssignmentResp)
+		if err != nil {
+			return nil, err
+		}
+
+		accountPermissions := utils.PathSearch("account_assignments", getAccountAssignmentRespBody, nil)
+
+		if accountPermissions == nil {
+			return nil, fmt.Errorf("error get Identity Center account assignment")
+		}
+
+		for _, v := range accountPermissions.([]interface{}) {
+			if principalId == utils.PathSearch("principal_id", v, "") {
+				accountPermission = v
+				break getAccountPermissionsLoop
```

Comment:
we can return the accountPermission in there

Reply:
done